src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_read_obs.F90

```diff
@@ -1548,7 +1548,461 @@ subroutine read_obs_sm_ASCAT(                                  &
+    
+    nullify( tmp_obs, tmp_lat, tmp_lon, tmp_tile_num, tmp_jtime )    
+    
+    ! ---------------
+    
+    ! initialize
+    
+    found_obs = .false.
+    
+    ! find files that are within half-open interval 
+    !   [date_time-dtstep_assim/2,date_time+dtstep_assim/2)
+
+    date_time_low = date_time    
+    call augment_date_time( -(dtstep_assim/2), date_time_low)
+    date_time_up = date_time    
+    call augment_date_time(  (dtstep_assim/2), date_time_up)
+    
+    ! calculate "extra" date_time_low to catch files w/ time swath start stamps before window 
+    !   but containing relevant obs
+    
+    date_time_low_fname = date_time_low
+    call augment_date_time( -dt_ASCAT_obsfile, date_time_low_fname)
+    
+    ! ----------------------------------------------------------------
+    !
+    ! read file with list of ASCAT file names for first day
+    
+    fname_of_fname_list = 'dummy'  ! make sure it is properly defined in obs_param nml
+    
+    call read_obs_SMAP_fnames( date_time_low_fname, this_obs_param,            &
+         fname_of_fname_list, N_fnames_max,                                    &
+         N_fnames, fname_list(1:N_fnames_max) )
+    
+    ! if needed, read file with list of ASCAT file names for second day and add
+    !   file names into "fname_list"
+    
+    if (date_time_low_fname%day /= date_time_up%day) then
+       
+       call read_obs_SMAP_fnames( date_time_up, this_obs_param,                &
+            fname_of_fname_list, N_fnames_max,                                 &
+            N_fnames_tmp, fname_list((N_fnames+1):(N_fnames+N_fnames_max)) )
+       
+       N_fnames = N_fnames + N_fnames_tmp
+       
+    end if
+    
+    tmpfnames = fname_list(1:N_fnames)
+    
+    ! ----------------------------------------------------------------
+    !
+    ! find files that have obs within assimilation window
+    
+    N_tmp = 0   
+    
+    do kk = 1,N_fnames
+
+       tmpfname = fname_list(kk)
+
+       ! Are we in the required assimilation window?
+       !
+       ! e.g. Y2019/M07/M01-ASCA-ASCSMO02-NA-5.0-20190702075700.000000000Z-20190702084627-1350204.bfr
+       !
+       !      12345678901234567890123456789012345678901234567890123456789012345678901234567890 
+       !               1         2         3         4         5         6         7
+
+       str_date_time = tmpfname(40:53)
+      
+       read(str_date_time( 1: 4), *) date_time_tmp%year
+       read(str_date_time( 5: 6), *) date_time_tmp%month
+       read(str_date_time( 7: 8), *) date_time_tmp%day
+       read(str_date_time( 9:10), *) date_time_tmp%hour
+       read(str_date_time(11:12), *) date_time_tmp%min
+       read(str_date_time(13:14), *) date_time_tmp%sec
+       
+       if ( datetime_lt_refdatetime( date_time_low_fname, date_time_tmp ) .and.          &
+            datetime_le_refdatetime( date_time_tmp,       date_time_up  )       ) then 
```

Comment:
Are the "lt" and "le" comparisons consistent with the half-open interval `[date_time_low, date_time_up)` mentioned above?  Should this be "le"  first and then "lt"?

Reply:
Fixed comment to reflect what we're actually doing with the half-open interval

https://github.com/GEOS-ESM/GEOSldas/blob/3f95ee02780cf30259b9a2ea85eceb000822a7e7/src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_read_obs.F90#L1668C1-L1669C60

Reply:
Change half-open interval comment
```
    ! find files that are within half-open interval 
    !   (date_time-dtstep_assim/2,date_time+dtstep_assim/2]
```